feat: Add HDR4x4 and HDR6x6 samples

hybridherbst opened the pull request:
Samples I made for https://github.com/mrdoob/three.js/pull/31689.

I don't think `ktx create` can produce these formats right now; they were created with the basisu executable with default settings and mipmaps. Note the default settings "upscale" a LDR image like here to HDR at 100 nits, so the images appear brigther / overblown.

```bash
./basisu -hdr_4x4 basis_uastc_hdr4x4.png -mipmap
./basisu -hdr_6x6 basis_uastc_hdr6x6.png -mipmap
```

hybridherbst committed:
feat: Add HDR4x4 and HDR6x6 samples

donmccurdy commented:
Thanks @hybridherbst! Looking forward to including these examples. Later I think we could include both plain and Basis versions, perhaps "BASIS HDR 4X4", "BASIS HDR 6X6", "ASTC HDR 4x4" and "ASTC HDR 6X6". But no need to make changes in this PR.

I'd just like to wait until we have a response about the DFD Color Model in https://github.com/mrdoob/three.js/pull/31689#issuecomment-3204598141, since I suspect the current color model might be an encoder bug.  Perhaps related, `ktx info encoded_file_6x6_no_mips.ktx2` gives validation errors:

```
error-6028: Invalid sample in basic DFD block. The samples must match the expected samples of the VkFormat.
    DFD block #1 sample #1 qualifierSigned in basic DFD block is 0 but the expected value is 1 for VK_FORMAT_ASTC_6x6_SFLOAT_BLOCK.
error-6028: Invalid sample in basic DFD block. The samples must match the expected samples of the VkFormat.
    DFD block #1 sample #1 sampleLower in basic DFD block is 0 but the expected value is 3212836864 for VK_FORMAT_ASTC_6x6_SFLOAT_BLOCK.
```

We'll also want to skip the upscaling. I imagine there's some purpose for that with HDR photography, but for purposes of this repository I think we want to the textures going 0–1 so it's easy to verify output matches other textures.

hybridherbst commented:
Understood!
It might make sense to have actual EXR images as source as well, because I find it hard to confirm with an LDR image whether HDR loading has actually worked.

donmccurdy commented:
Affinity Designer can do EXR export, but there was some reason I'm doing PNG-to-EXR in the `encode.sh` script that I can't recall at the moment. But yes... starting from EXR would avoid the upscaling too, which would be nice.

hybridherbst commented:
Here's EXR RGB+ZIP versions of the base files for completeness. If Affinity can export them it's probably cleaner for future changes – these two are converted with Blender (my go-to tool for image file conversions by now, because it has the various ASWF tools built in).

[basis_uastc_hdr_exr.zip](https://github.com/user-attachments/files/21899801/basis_uastc_hdr_exr.zip)

hybridherbst committed:
remove 6x6 for now since it seems to be invalid

hybridherbst commented:
I updated the 4x4 image without upconversion scaling and removed the invalid 6x6 image for now.

```
./basisu -hdr_4x4 basis_uastc_hdr4x4.png -mipmap -hdr_ldr_upconversion_nit_multiplier 1
```

hybridherbst committed:
bring back hdr6x6 and hdr6x6i as they seem to be valid after all

hybridherbst commented:
After updating basis to 1.6, the `2d_astc_hdr6x6.ktx2` file both loads and transcodes just fine – so I think it's valid after all.
I now also added a `2d_astc_hdr6x6i.ktx2` (intermediate) variant. This one doesn't load yet though, it seems that ktx-parse errors out.
